src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -2,48 +2,2208 @@
+            Assert.Equal(comp.SyntaxTrees[1], cs[1].DeclaringSyntaxReferences.Single().SyntaxTree);
+
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: true);
+            var cReference = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().Last();
+            var info = model.GetTypeInfo(cReference);
+            Assert.Equal(expectedSymbol.GetPublicSymbol(), info.Type);
+        }
+    }
+
+    [Theory]
+    [InlineData("file", "file")]
+    [InlineData("file", "")]
+    [InlineData("", "file")]
+    public void Duplication_11(string firstFileModifier, string secondFileModifier)
+    {
+        var source1 = $$"""
+            using System;
+
+            {{firstFileModifier}} partial class Outer
+            {
+                internal class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(1);
+                    }
+                }
+            }
+            """;
+
+        var source2 = $$"""
+            using System;
+
+            {{secondFileModifier}} partial class Outer
+            {
+                internal class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(2);
+                    }
+                }
+            }
+            """;
+
+        var main = """
+            class Program
+            {
+                static void Main()
+                {
+                    Outer.C.M();
+                }
+            }
+            """;
+
+        // PROTOTYPE(ft): execute and check expectedOutput once name mangling is done
+        // expectedOutput: "1"
+        var comp = CreateCompilation(new[] { source1 + main, source2 });
+        var outers = comp.GetMembers("Outer");
+        var cs = outers.Select(o => ((NamedTypeSymbol)o).GetMember("C")).ToArray();
+        var tree = comp.SyntaxTrees[0];
+        var expectedSymbol = cs[0];
+        verify();
+
+        // expectedOutput: "2"
+        comp = CreateCompilation(new[] { source1, source2 + main });
+        outers = comp.GetMembers("Outer");
+        cs = outers.Select(o => ((NamedTypeSymbol)o).GetMember("C")).ToArray();
+        tree = comp.SyntaxTrees[1];
+        expectedSymbol = cs[1];
+        verify();
+
+        void verify()
+        {
             comp.VerifyDiagnostics();
+            Assert.Equal(2, cs.Length);
+            Assert.Equal(comp.SyntaxTrees[0], cs[0].DeclaringSyntaxReferences.Single().SyntaxTree);
+            Assert.Equal(comp.SyntaxTrees[1], cs[1].DeclaringSyntaxReferences.Single().SyntaxTree);
+
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: true);
+            var cReference = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().Last();
+            var info = model.GetTypeInfo(cReference);
+            Assert.Equal(expectedSymbol.GetPublicSymbol(), info.Type);
         }
     }
```

Comment:
Should we test `base` references, in error scenarios?

```csharp
// file1.cs
file class Base
{
    public Base(int i) { }
}
partial class Derived : Base
{
}

// file2.cs
partial class Derived
{
    Derived() : base(0) { }
}
```

```csharp
// file1.cs
file class Base
{
    public virtual void F1() { }
}
partial class Derived : Base
{
    public override void F1() { }
}

// file2.cs
partial class Derived
{
    void F2() { base.F1(); }
}
```

Reply:
Added a line to #60819.